hhs_facilities/delphi_hhs_facilities/geo.py

```diff
@@ -0,0 +1,37 @@
+"""Functions for mapping geographic regions."""
+
+import pandas as pd
+from delphi_utils.geomap import GeoMapper
+
+
+def convert_geo(df: pd.DataFrame, geo: str, gmpr: GeoMapper) -> pd.DataFrame:
+    """
+    Map a DataFrame to desired regions.
+
+    The HHS facility level data contains a zip, state, and fips columns. For state and fips, we
+    use them as given. For all other geos, we map from zip (the smallest of the regions) to the
+    desired geo.
```

Comment:
> For all other geos, we map from zip

Some zips cross county and state lines, so the geomapper will split those entries proportionally by population. That's probably not what we want, unless we have reason not to trust the FIPS published by HHS.

Are there facilities that provide a zip but no fips?

Reply:
> Are there facilities that provide a zip but no fips?
Good question, here's what I get:
```
In [5]: sum(pd.isna(df.zip))                                                                                                                                         
Out[5]: 9

In [6]: sum(pd.isna(df.fips_code))                                                                                                                                   
Out[6]: 239
```

I wonder if we should do a more custom mapping, e.g. for HRR do zip -> HRR and fips -> state since those nest properly. Not sure if we want to map the missing fips from the zips

Reply:
> Not sure if we want to map the missing fips from the zips

It's a matter of what the most reliable guess is. If the FIPS is included by the source, that's the best estimate of the actual FIPS. If the FIPS is missing, then the geo utility's estimate(s) from ZIP5 are the best guess.

Reply:
So probably do a preprocessing pass to fill in missing FIPS and then proceed with your approach

Reply:
ok that sounds good to me. I've implemented the zip->HRR and fips->else logic in this PR, and will follow up with a separate PR to fill in missing FIPS